### PR TITLE
fix(api): if needed, home both pipettes with InstrumentContext.home

### LINKED ISSUE #7499: [bug] After switching to other pipette, current pipette fails to home if other was just homed.
I'm using the Jupyter notebook to test movement commands on API v2.9.

The following code creates unexpected (at least to me) and dangerous behaviour, that is the previously used pipette stays at its current Z height instead of homing before movement.

```python
import opentrons.execute
from opentrons.types import Point, Location
protocol_context = opentrons.execute.get_protocol_api('2.9')

pipette_20 = protocol_context.load_instrument(
        "p20_single_gen2", mount='left', tip_racks=[])
pipette_300 = protocol_context.load_instrument(
        "p300_single_gen2", mount='right', tip_racks=[])

protocol_context.home()
pipette_20.move_to(Location(Point(224,230,139.9), None), speed=100, minimum_z_height=130)

# this line is causing the erroneous behaviour during the next line
pipette_300.home() 

# This line normally would first home the "20 ul pipette" before moving, however due to the .home() previously, it forgets to do so
pipette_300.move_to(Location(Point(224,230,139.9), None), speed=100, minimum_z_height=130)
```

At the end of this, both pipettes are at a "low" position, which as far as I understand should never happen.

Server version: 4.1.1
Firmware Version: v1.1.0-25e5cea

As a related feature request, it would be great if during normal behaviour, instead of fully "homing" the "other" pipette (which takes forever to check with the limit switches), it would simply raise it - I understand it's better to check everything from hardware instead of software, but it would be great to have a "dangerous" mode for people who like to go fast :)

### PR DESCRIPTION
**Pull request by mcous**
## Overview

When using `HardwareControlAPI.home_z` (by way of `InstrumentContext.home`), the HW API can be put into a state where it thinks both pipettes are retracted, when in reality, only one pipette is retracted. In this state, movements become very likely to cause collisions.

Fixes #7499

## Changelog

Now, when calling `InstrumentContext.home`, 

- fix(api): if needed, home both pipettes with InstrumentContext.home

## Review requests

I tested the following behaviors on an OT-2R in the office. I think given these smoke tests + unit tests, a code review from the embedded team is all that's left for this PR to be good to merge.

- [x] Protocol API versions at or below `2.12` exhibit buggy behavior when running repro protocol in 7499
- [x] Protocol API versions at or above `2.13` avoid buggy behavior by homing both pipettes if needed during `pipette.home()`

## Risk assessment

Low; Unit test coverage added and g-code regression tests appear to be capturing this behavioral change.

I updated the G-code regression test snapshots after careful examination with @sfoster1. It turns out that `swift_turbo.py` and `swift_smoke.py` had the potentially buggy behavior, and this PR fixed the bug.

### CODE REVIEW

**Review by sfoster1 — DISMISSED** [review]
Not sure about the api level fix - this fixes a bug that affects motion, but the workaround would be unaffected after the pr is merged. It is true, however, that a protocol that would work after this fix might not work before it. I suppose we could bias towards an API level bump.

The best place for a test would be `tests/opentrons/hardware_control/test_moves.py` - thanks!

**Comment by mcous** [comment]
Looks like we got some g-code test failures (among other things). I've got some unit tests cooking, and I think a PAPI bump is the smart thing to do.

Will put this PR back into a draft when I'm back at a computer to fix up

File: api/src/opentrons/protocols/context/protocol_api/instrument_context.py [hunk 1]
```diff
@@ -122,7 +122,12 @@ def drop_tip(self, home_after: bool) -> None:
 
     def home(self) -> None:
         """Home the mount"""
-        self._protocol_interface.get_hardware().home_z(self._mount)
+        self._protocol_interface.get_hardware().home_z(
+            mount=self._mount,
+            # preserve buggy behavior in <= 2.12 strict back. compat.
```
Comment (mcous) [inline api/src/opentrons/protocols/context/protocol_api/instrument_context.py]:
```suggestion
            # preserve buggy behavior in <= 2.12 for strict back. compat.
```

**Review by sfoster1 — APPROVED** [review]
Looks good to me, behavior makes sense. On the OT-3 we'll want some more complex behavior since we have encoders on these axes, so I'm glad you just dropped a todo there.

**Pull request merged**